slight/src/commands/run.rs

```diff
@@ -147,132 +147,87 @@ fn build_store_instance(
-                    builder.link_capability::<Pubsub>("pubsub".to_string())?;
+                    builder.link_capability::<Pubsub>()?;
                     linked_capabilities.insert("pubsub".to_string());
                 }
 
-                maybe_add_named_capability_to_store(
-                    &toml.specversion,
-                    toml.secret_store.clone(),
-                    &mut capability_store,
-                    c.clone(),
-                    resource_map.clone(),
-                    &toml_file_path,
-                )?;
-
-                slight_builder =
-                    slight_builder.add_state(State::PubSub(slight_pubsub::PubsubState::new(
-                        resource_type.to_string(),
-                        capability_store.clone(),
-                    )))?;
+                let resource =
+                    slight_pubsub::Pubsub::new(resource_type.to_string(), capability_store.clone());
+                builder.add_to_builder("pubsub".to_string(), resource);
             }
             _ if CONFIGS_HOST_IMPLEMENTORS.contains(&resource_type) => {
                 if !linked_capabilities.contains("configs") {
-                    builder.link_capability::<Configs>("configs".to_string())?;
+                    builder.link_capability::<Configs>()?;
                     linked_capabilities.insert("configs".to_string());
                 }
 
-                maybe_add_named_capability_to_store(
-                    &toml.specversion,
-                    toml.secret_store.clone(),
-                    &mut capability_store,
-                    c.clone(),
-                    resource_map.clone(),
-                    &toml_file_path,
-                )?;
-
-                slight_builder = slight_builder.add_state(State::RtCfg(
-                    slight_runtime_configs::ConfigsState::new(
-                        resource_type.to_string(),
-                        capability_store.clone(),
-                    ),
-                ))?;
+                let resource = slight_runtime_configs::Configs::new(
+                    resource_type.to_string(),
+                    capability_store.clone(),
+                );
+                builder.add_to_builder("configs".to_string(), resource);
             }
             "http" => {
                 if !linked_capabilities.contains("http") {
-                    builder.link_capability::<Http<Builder>>(resource_type.to_string())?;
-                    linked_capabilities.insert("http".to_string());
-
-                    slight_builder = slight_builder.add_state(State::Http(
-                        slight_http::HttpState::<Builder>::new(resource_map.clone()),
-                    ))?;
+                    let http = slight_http::Http::<Builder>::new(resource_map.clone());
+                    builder
+                        .link_capability::<Http<Builder>>()?
+                        .add_to_builder("http".to_string(), http);
```

Comment:
```suggestion
                    let http = slight_http::Http::<Builder>::new(resource_map.clone());
                    builder
                        .link_capability::<Http<Builder>>()?
                        .add_to_builder("http".to_string(), http);
                    linked_capabilities.insert("http".to_string());
```

Reply:
wooo I forgot adding back this line! Thanks